llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -2179,11 +2180,23 @@ MemoryDepChecker::Dependence::DepType MemoryDepChecker::isDependent(
   // If MinNumIter is 4 (Say if a user forces the vectorization factor to be 4),
   // the minimum distance needed is 28, which is greater than distance. It is
   // not safe to do vectorization.
+
+  // We know that Dist is positive, but it may not be constant. Use the signed
+  // minimum for computations below, as this ensures we compute the closest
+  // possible dependence distance.
+  int64_t MinDistance = SE.getSignedRangeMin(Dist).getSExtValue();
```

Comment:
What happens if MinDistance is negative?  (The code claims we "know" it's positive, but does SCEV always know that?)

Reply:
We bail out with `if (!SE.isKnownPositive(Dist))` (line 2123), but that might not be sufficient?

Reply:
isKnownPositive() is a wrapper around getSignedRangeMin(), so that's probably fine; maybe refactor to make it more explicit.

Reply:
Adjusted to first get MinDistance and then check that value, thanks!